src/transformers/models/swiftformer/modeling_swiftformer.py

```diff
@@ -0,0 +1,726 @@
+            x = input + self.drop_path(x)
+        return x
+
+
+class Mlp(nn.Module):
+    """
+    Implementation of MLP layer with 1*1 convolutions. Input: tensor with shape [B, C, H, W] Output: tensor with shape
+    [B, C, H, W]
+    """
+
+    def __init__(self, in_features, hidden_features=None, out_features=None, act_layer=nn.GELU, drop=0.0):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        self.norm1 = nn.BatchNorm2d(in_features)
+        self.fc1 = nn.Conv2d(in_features, hidden_features, 1)
+        self.act = act_layer()
+        self.fc2 = nn.Conv2d(hidden_features, out_features, 1)
+        self.drop = nn.Dropout(drop)
+        self.apply(self._init_weights)
+
+    def _init_weights(self, m):
+        if isinstance(m, nn.Conv2d):
+            trunc_normal_(m.weight, std=0.02)
+            if m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+
+    def forward(self, x):
+        x = self.norm1(x)
+        x = self.fc1(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+
+class EfficientAdditiveAttention(nn.Module):
+    """
+    Efficient Additive Attention module for SwiftFormer. Input: tensor in shape [B, C, H, W] Output: tensor in shape
+    [B, C, H, W]
+    """
+
+    def __init__(self, in_dims=512, token_dim=256, num_heads=2):
+        super().__init__()
+
+        self.to_query = nn.Linear(in_dims, token_dim * num_heads)
+        self.to_key = nn.Linear(in_dims, token_dim * num_heads)
+
+        self.w_g = nn.Parameter(torch.randn(token_dim * num_heads, 1))
+        self.scale_factor = token_dim**-0.5
+        self.Proj = nn.Linear(token_dim * num_heads, token_dim * num_heads)
+        self.final = nn.Linear(token_dim * num_heads, token_dim)
+
+    def forward(self, x):
+        query = self.to_query(x)
+        key = self.to_key(x)
+
+        query = torch.nn.functional.normalize(query, dim=-1)
+        key = torch.nn.functional.normalize(key, dim=-1)
+
+        query_weight = query @ self.w_g
+        A = query_weight * self.scale_factor
+
+        A = A.softmax(dim=-1)
+
+        G = torch.sum(A * query, dim=1)
+
+        G = G.unsqueeze(1).repeat(1, key.shape[1], 1)
+
+        out = self.Proj(G * key) + query
+
+        out = self.final(out)
+
+        return out
+
+
+class SwiftFormerLocalRepresentation(nn.Module):
+    """
+    Local Representation module for SwiftFormer that is implemented by 3*3 depth-wise and point-wise convolutions.
+    Input: tensor in shape [B, C, H, W] Output: tensor in shape [B, C, H, W]
+    """
+
+    def __init__(self, dim, kernel_size=3, drop_path=0.0, use_layer_scale=True):
+        super().__init__()
+        self.dwconv = nn.Conv2d(dim, dim, kernel_size=kernel_size, padding=kernel_size // 2, groups=dim)
+        self.norm = nn.BatchNorm2d(dim)
+        self.pwconv1 = nn.Conv2d(dim, dim, kernel_size=1)
```

Comment:
I'd rename to make more explicit - same with second conv layer

```suggestion
        self.point_wise_conv1 = nn.Conv2d(dim, dim, kernel_size=1)
```